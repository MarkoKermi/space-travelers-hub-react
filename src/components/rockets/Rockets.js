import { Badge, Button, Card } from 'react-bootstrap';
import { PropTypes } from 'prop-types';
import { useDispatch } from 'react-redux';
import styles from './RocketsFalcon.module.css';
import { bookRockets } from '../../redux/rockets/rocketsSlice';

const Rockets = ({
  id, name, description, image, isReserved,
}) => {
  const dispatch = useDispatch();
  return (
    <Card className="d-flex flex-row m-5">
      <Card.Img className={styles.rocketImage} variant="top" src={image} />
      <Card.Body className="d-flex flex-column">
        <h4 className=" card-title text-start p-1 ">{name}</h4>
        {isReserved && <Badge className="align-self-start" bg="info">Reserved</Badge>}
        <Card.Text className=" text-start p-1 ">{description}</Card.Text>
        <Button
          className="align-self-start"
          variant={isReserved ? 'outline-secondary' : 'primary'}
          onClick={() => dispatch(bookRockets(id))}
        >
          {isReserved ? 'Cancel Reservation' : 'Reserve rocket'}
        </Button>
      </Card.Body>
    </Card>
  );
};

Rockets.defaultProps = {
  id: '', name: '', description: '', image: '', isReserved: false,
};
Rockets.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string,
  description: PropTypes.string,
  image: PropTypes.string,
  isReserved: PropTypes.bool,
};

export default Rockets;
