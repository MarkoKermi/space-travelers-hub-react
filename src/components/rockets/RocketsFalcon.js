import Button from 'react-bootstrap/Button';
import Card from 'react-bootstrap/Card';
import styles from './RocketsFalcon.module.css';

const RocketsFalcon = () => (
  <Card className={styles.rocketFalcon} style={{ width: '18rem' }}>
    <Card.Img variant="top" src="holder.js/100px180" />
    <Card.Body>
      <Card.Title>Falcon 1</Card.Title>
      <Card.Text>
        Some quick example text to build on the card title and make up the
        bulk of the card content.
      </Card.Text>
      <Button variant="primary">Reserve Rocket</Button>
    </Card.Body>
  </Card>
);

export default RocketsFalcon;
