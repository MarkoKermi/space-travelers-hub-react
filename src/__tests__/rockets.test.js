import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Rockets from '../components/rockets/Rockets';
import RocketsFalcon from '../components/rockets/RocketsFalcon';

describe('testing snapshots of Rocket components', () => {
  it('testing Rocket component', () => {
    const rocket = renderer.create(<Provider store={store}><Rockets /></Provider>).toJSON();
    expect(rocket).toMatchSnapshot();
  });
  it('testing Rockets Falcon component', () => {
    const rocketFalcon = renderer.create(
      <Provider store={store}>
        <RocketsFalcon />
      </Provider>,
    ).toJSON();
    expect(rocketFalcon).toMatchSnapshot();
  });
});
