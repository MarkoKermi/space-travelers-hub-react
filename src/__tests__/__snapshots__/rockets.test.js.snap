// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing snapshots of Rocket components testing Rocket component 1`] = `
<div
  className="d-flex flex-row m-5 card"
>
  <img
    className="card-img-top rocketImage"
    src=""
  />
  <div
    className="d-flex flex-column card-body"
  >
    <h4
      className="card-title text-start p-1 "
    />
    <p
      className=" text-start p-1  card-text"
    />
    <button
      className="align-self-start btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Reserve rocket
    </button>
  </div>
</div>
`;

exports[`testing snapshots of Rocket components testing Rockets Falcon component 1`] = `<div />`;
